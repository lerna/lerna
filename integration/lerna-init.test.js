"use strict";

const loadJsonFile = require("load-json-file");
const path = require("path");
const tempy = require("tempy");

const cliRunner = require("@lerna-test/cli-runner");
const initFixture = require("@lerna-test/init-fixture")(__dirname);

describe("lerna init", () => {
  const parsePackageJson = cwd => loadJsonFile(path.join(cwd, "package.json"));
  const parseLernaJson = cwd => loadJsonFile(path.join(cwd, "lerna.json"));
  const loadMetaData = cwd => Promise.all([parsePackageJson(cwd), parseLernaJson(cwd)]);

  test("initializes empty directory", async () => {
    const cwd = tempy.directory();

    const { stderr } = await cliRunner(cwd)("init");
    expect(stderr).toMatchInlineSnapshot(`
      lerna notice cli __TEST_VERSION__
      lerna info Initializing Git repository
      lerna info Creating package.json
      lerna info Creating lerna.json
      lerna info Creating packages directory
      lerna success Initialized Lerna files
    `);

    const [packageJson, lernaJson] = await loadMetaData(cwd);
    expect(packageJson).toMatchInlineSnapshot(`
      Object {
        "devDependencies": Object {
          "lerna": "^__TEST_VERSION__",
        },
        "name": "root",
        "private": true,
      }
    `);
    expect(lernaJson).toMatchInlineSnapshot(`
      Object {
        "packages": Array [
          "packages/*",
        ],
        "version": "0.0.0",
      }
    `);
  });

  test("updates existing metadata", async () => {
    const cwd = await initFixture("lerna-init");

    const { stderr } = await cliRunner(cwd)("init", "--exact");
    expect(stderr).toMatchInlineSnapshot(`
      lerna notice cli __TEST_VERSION__
      lerna WARN project Deprecated key "commands" found in lerna.json
      lerna WARN project Please update "commands" => "command"
      lerna info Updating package.json
      lerna info Updating lerna.json
      lerna info Creating packages directory
      lerna success Initialized Lerna files
    `);

    const [packageJson, lernaJson] = await loadMetaData(cwd);
    expect(packageJson).toMatchInlineSnapshot(`
      Object {
        "devDependencies": Object {
          "lerna": "__TEST_VERSION__",
        },
        "name": "updates",
      }
    `);
    expect(lernaJson).toMatchInlineSnapshot(`
      Object {
        "command": Object {
          "bootstrap": Object {
            "hoist": true,
          },
          "init": Object {
            "exact": true,
          },
        },
        "packages": Array [
          "packages/*",
        ],
        "version": "1.0.0",
      }
    `);
  });
});
