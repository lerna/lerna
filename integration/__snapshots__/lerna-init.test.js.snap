// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lerna init initializes empty directory: lerna.json 1`] = `
Object {
  packages: Array [
    packages/*,
  ],
  version: 0.0.0,
}
`;

exports[`lerna init initializes empty directory: package.json 1`] = `
Object {
  devDependencies: Object {
    lerna: "^__TEST_VERSION__",
  },
  name: root,
  private: true,
}
`;

exports[`lerna init initializes empty directory: stderr 1`] = `
lerna info version __TEST_VERSION__
lerna info Initializing Git repository
lerna info Creating package.json
lerna info Creating lerna.json
lerna info Creating packages directory
lerna success Initialized Lerna files
`;

exports[`lerna init updates existing metadata: lerna.json 1`] = `
Object {
  command: Object {
    bootstrap: Object {
      hoist: true,
    },
    init: Object {
      exact: true,
    },
  },
  packages: Array [
    packages/*,
  ],
  version: 1.0.0,
}
`;

exports[`lerna init updates existing metadata: package.json 1`] = `
Object {
  devDependencies: Object {
    lerna: "__TEST_VERSION__",
  },
  name: updates,
}
`;

exports[`lerna init updates existing metadata: stderr 1`] = `
lerna info version __TEST_VERSION__
lerna WARN project Deprecated key "commands" found in lerna.json
lerna WARN project Please rename "commands" => "command"
lerna info Updating package.json
lerna info Updating lerna.json
lerna info Creating packages directory
lerna success Initialized Lerna files
`;
